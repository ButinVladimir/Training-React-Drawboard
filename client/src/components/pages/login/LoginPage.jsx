import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

const LoginPage = ({
  locked,
  errorMessage,
  room,
  onSetRoom,
  onLogin,
}) => (
  <div className="login-container">
    <div className="row">
      <div className="col header">Room:</div>
      <div className="col">
        <input className="input" value={room} onChange={e => onSetRoom(e.target.value)} />
      </div>
    </div>
    {errorMessage && (
      <div className="row">
        <div className="col">
          {errorMessage}
        </div>
      </div>
    )}
    <div className="row right">
      <div className="col">
        <button type="button" disabled={locked} onClick={onLogin}>Log in</button>
      </div>
    </div>
  </div>
);

LoginPage.propTypes = {
  locked: PropTypes.bool.isRequired,
  errorMessage: PropTypes.string,
  room: PropTypes.string.isRequired,
  onSetRoom: PropTypes.func.isRequired,
  onLogin: PropTypes.func.isRequired,
};

LoginPage.defaultProps = {
  errorMessage: '',
};

export default LoginPage;
