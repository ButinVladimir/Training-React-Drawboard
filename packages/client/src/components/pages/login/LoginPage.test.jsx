import React from 'react';
import { shallow } from 'enzyme';
import LoginPage from './LoginPage';

describe('LoginPage', () => {
  it('renders properly', () => {
    const component = shallow(<LoginPage
      locked={false}
      errorMessage=""
      room=""
      onSetRoom={() => {}}
      onLogin={() => {}}
    />);

    expect(component).toMatchSnapshot();
  });

  it('renders properly when form is locked', () => {
    const component = shallow(<LoginPage
      locked
      errorMessage=""
      room=""
      onSetRoom={() => {}}
      onLogin={() => {}}
    />);

    expect(component).toMatchSnapshot();
  });

  it('renders properly when form has error', () => {
    const component = shallow(<LoginPage
      locked={false}
      errorMessage="An error occurred"
      room=""
      onSetRoom={() => {}}
      onLogin={() => {}}
    />);

    expect(component).toMatchSnapshot();
  });

  it('renders properly when room has value', () => {
    const component = shallow(<LoginPage
      locked={false}
      errorMessage=""
      room="The Room"
      onSetRoom={() => {}}
      onLogin={() => {}}
    />);

    expect(component).toMatchSnapshot();
  });

  it('handles onSetRoom event', () => {
    const handler = jest.fn();
    const component = shallow(<LoginPage
      locked={false}
      errorMessage=""
      room="The Room"
      onSetRoom={handler}
      onLogin={() => {}}
    />);

    component.find('input[name="room"]').simulate('change', { target: { value: 'value' } });

    expect(handler).toHaveBeenCalledTimes(1);
    expect(handler).toHaveBeenCalledWith('value');
  });

  it('handles onLogin event', () => {
    const handler = jest.fn();
    const component = shallow(<LoginPage
      locked={false}
      errorMessage=""
      room="The Room"
      onSetRoom={() => {}}
      onLogin={handler}
    />);

    component.find('button[name="enter"]').simulate('click');

    expect(handler).toHaveBeenCalled();
  });
});
