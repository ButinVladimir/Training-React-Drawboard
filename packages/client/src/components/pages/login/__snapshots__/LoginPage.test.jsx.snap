// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage renders properly 1`] = `
<div
  className="login-container"
>
  <div
    className="row"
  >
    <div
      className="col header"
    >
      Room:
    </div>
    <div
      className="col"
    >
      <input
        className="input"
        name="room"
        onChange={[Function]}
        value=""
      />
    </div>
  </div>
  <div
    className="row right"
  >
    <div
      className="col"
    >
      <button
        disabled={false}
        name="enter"
        onClick={[Function]}
        type="button"
      >
        Log in
      </button>
    </div>
  </div>
</div>
`;

exports[`LoginPage renders properly when form has error 1`] = `
<div
  className="login-container"
>
  <div
    className="row"
  >
    <div
      className="col header"
    >
      Room:
    </div>
    <div
      className="col"
    >
      <input
        className="input"
        name="room"
        onChange={[Function]}
        value=""
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      An error occurred
    </div>
  </div>
  <div
    className="row right"
  >
    <div
      className="col"
    >
      <button
        disabled={false}
        name="enter"
        onClick={[Function]}
        type="button"
      >
        Log in
      </button>
    </div>
  </div>
</div>
`;

exports[`LoginPage renders properly when form is locked 1`] = `
<div
  className="login-container"
>
  <div
    className="row"
  >
    <div
      className="col header"
    >
      Room:
    </div>
    <div
      className="col"
    >
      <input
        className="input"
        name="room"
        onChange={[Function]}
        value=""
      />
    </div>
  </div>
  <div
    className="row right"
  >
    <div
      className="col"
    >
      <button
        disabled={true}
        name="enter"
        onClick={[Function]}
        type="button"
      >
        Log in
      </button>
    </div>
  </div>
</div>
`;

exports[`LoginPage renders properly when room has value 1`] = `
<div
  className="login-container"
>
  <div
    className="row"
  >
    <div
      className="col header"
    >
      Room:
    </div>
    <div
      className="col"
    >
      <input
        className="input"
        name="room"
        onChange={[Function]}
        value="The Room"
      />
    </div>
  </div>
  <div
    className="row right"
  >
    <div
      className="col"
    >
      <button
        disabled={false}
        name="enter"
        onClick={[Function]}
        type="button"
      >
        Log in
      </button>
    </div>
  </div>
</div>
`;
